server/datastore/cached_mysql/cached_mysql_test.go

```diff
@@ -80,6 +81,31 @@ func TestClone(t *testing.T) {
 			clone, err := clone(tc.src)
 			require.NoError(t, err)
 			assert.Equal(t, tc.want, clone)
+
```

Comment:
Did we miss a test case with `AppConfig` in 
```
	tests := []struct {
		name string
		src  interface{}
		want interface{}
	}{
```
that would have caught this issue?

Reply:
The problem with those cases is that they are all simple values to clone, no structs. I'll try to add one struct to reproduce.

Reply:
Ha, just noticed I had a pending review in progress that I didn't submit, with a comment explaining the new tests I added...

Reply:
Added a case that reproduces the issue in question (and passes with deep clone).